CP2A-BucketList-Application-API

asce1062 opened the pull request:
**What does this PR do?**

An API for an online Bucket List service using Flask. 


**Description of Task to be completed?**

### Problem Description

According to Merriam-Webster Dictionary, a Bucket List is a list of things that one has not done before but wants to do before dying.
In this exercise you will be required to create an API for an online Bucket List service using Flask.
Specifications for the API are shown below. You may use any database you prefer for this assignment.


EndPoint    |   Functionality
:-----------|:---------------
`POST /auth/login`                          | Logs a user in
`POST /auth/register`                       | Register a user
`POST /bucketlists/`                        | Logs a user in
`GET /bucketlists/`                         | List all the created bucket lists
`GET /bucketlists/<id>`                     | Get single bucket list
`PUT /bucketlists/<id>`                     | Update this bucket list
`DELETE /bucketlists/<id>`                  | Delete this single bucket list
`POST /bucketlists/<id>/items/`             | Create a new item in bucket list
`PUT /bucketlists/<id>/items/<item_id>`     | Update a bucket list item
`DELETE /bucketlists/<id>/items/<item_id>`  | Delete an item in a bucket list


**How should this be manually tested?**

`nosetests -v`

Commit:
[CHORE] Initialize file structure.

Commit:
[CHORE] Update README with detailed problem description

Commit:
[CHORE] Update README

Commit:
[CHORE] Environment configurations, Create the app, Data Model, Making Migrations

Commit:
[CHORE] Add Bucketlist(s) tests

Commit:
[CHORE] Add User authentication tests

Commit:
[CHORE] Add bucketlist(s) item(s) tests

Commit:
[CHORE] Cleanup/Beautify code

Commit:
[CHORE] Add comments to improve code readability

Commit:
[CHORE] Add User resource

Add user login and register resources

Commit:
[CHORE] Add Tests to cover the added user resources

Commit:
[CHORE] Add Bucketlist Resource

Commit:
[CHORE] Styling and Cleanup

Commit:
[FIX] Authorization token issue in tests, and start refactoring test

Commit:
[CHORE] Add more BucketListAPI tests

Commit:
[FIX] Get a single item by it's ID

Commit:
[CHORE] Add more tests

Commit:
[CHORE] Cleanup code

Commit:
[CHORE] Start working on an automation script

Commit:
[CHORE] Update automation script

munala commented on app/bucket_list.py:
```diff
@@ -0,0 +1,367 @@
+                return (
+                    {
+                        'message': 'Bucketlist not found'
+                    }, 400
```
Shouldn't this be 404?

munala commented on app/bucket_list.py:
```diff
@@ -0,0 +1,367 @@
+        if not this_bucket_list:
+            return (
+                {
+                    'error': 'bucketlist with id {} does not exists'.format(id)
```
Shouldn't it be `does not exist`?

munala commented on manage.py:
```diff
@@ -0,0 +1,43 @@
+def create_db():
+    """Creates database with tables"""
+
+    os.system('createdb flask_api')
```
I like this

munala commented:
Good stuff.

Commit:
[CHORE] Implement peer review feedback and rename some tests

Commit:
[FIX] Naming issue

Commit:
[CHORE] Checkpoint #1

Fix styling in about script. Add require key input for tests options in menu script .Add more comments (Highway to self documentation). Remove unused libraries from requirements.

Commit:
[CHORE] Update README #1

Commit:
[CHORE] Update README #1

Commit:
[CHORE] Update README #1

Commit:
[CHORE] Update README #1

Commit:
[CHORE] Update README #1

Commit:
[CHORE] Code cleanup

Commit:
[CHORE] Add badges

Commit:
[CHORE] Add Contribution badge

Commit:
[CHORE] Update README

Commit:
[CHORE] Remove repetition on README

Commit:
[CHORE] Update README with screenshots

Commit:
* Add .env_example
    - rename to .env ($ cp .env_example .env)
    - remove quotes
    - APP_SETTINGS takes development, testing, staging and production
        parametres
* Update .gitignore
    - do not ignore instance directory. It contains configs
* Fix tests
    - disable SQLALCHEMY_TRACK_MODIFICATIONS
* Add missing requirements
    - Nose
    - Coverage
